src/librustc_trans/cabi_x86_64.rs

```diff
@@ -175,18 +175,18 @@ fn reg_component(cls: &[Class], i: &mut usize, size: Size) -> Option<Reg> {
     }
 }
 
-fn cast_target(cls: &[Class], size: Size) -> CastTarget {
+fn cast_target(cls: &[Class], size: Size) -> Option<CastTarget> {
     let mut i = 0;
-    let lo = reg_component(cls, &mut i, size).unwrap();
+    let lo = reg_component(cls, &mut i, size)?;
     let offset = Size::from_bytes(8) * (i as u64);
     let target = if size <= offset {
         CastTarget::from(lo)
     } else {
-        let hi = reg_component(cls, &mut i, size - offset).unwrap();
+        let hi = reg_component(cls, &mut i, size - offset)?;
```

Comment:
These don't seem right - if the `unwrap` can fail that means there's a logic error somewhere within this function, which this seems to hide.

Reply:
Updated to call `bug!()` when this failed.